server/source/world/TerrainGenerator.hpp

```diff
@@ -48,6 +48,8 @@ class TerrainGenerator {
 		static float noise2d(double x, double y, int octaves, float persistence);
 		static float noise3d_abs(double x, double y, double z, int octaves, float persistence);
 
+		TerrainBiomeSampler biomeSampler;
+
 		u16 m_dirtBlockID = 0;
 		u16 m_grassBlockID = 0;
 		u16 m_stoneBlockID = 0;
```

Comment:
Are `m_dirtBlockID`, `m_grassBlockID` still relevant? If they're not, you can remove them.

Reply:
m_grassBlockID is helpful to not place tallgrass on stone or dirt, but removed m_dirt and m_stone.